src/lib/utitlities/synchronization.ts

```diff
@@ -0,0 +1,306 @@
+        if (remoteFolder.id in map) {
+          const localFolder = localMap[remoteFolder.id]
+          if (remoteFolder.updatedAt === localFolder.updatedAt) {
+            map[remoteFolder.id] = 'synchronized'
+          } else {
+            map[remoteFolder.id] = 'conflicted'
+          }
+        } else {
+          map[remoteFolder.id] = 'remote-only'
+        }
+      }
+      return map
+    }
+  )
+
+  const localMap = derived(
+    localStore,
+    (localFolders) => {
+      const map: FolderMap = {}
+      for (const localFolder of localFolders) {
+        map[localFolder.id] = localFolder
+      }
+      return map
+    }
+  )
+
+  const remoteMap = derived(
+    remoteStore,
+    (remoteFolders) => {
+      const map: FolderMap = {}
+      for (const remoteFolder of remoteFolders) {
+        map[remoteFolder.id] = remoteFolder
+      }
+      return map
+    }
+  )
+
+  return {
+    subscribe: store.subscribe,
+    localMap,
+    remoteMap,
+  }
+}
+
+/**
+ * Create one store that combine the data of local and remote folders. It also
+ * "auto merge" the data following "last write wins" rule.
+ *
+ * In more details:
+ *
+ * - If one folder exists within the local store but not within the remote
+ *   store, then the folder would be created within the remote store, and vice
+ *   versa.
+ * - If the folder within both stores, then the "newer" folder (one that has
+ *   greater `updatedAt` value) would be used and be written to both stores.
+ *
+ * This strategy is much simpler than `createHigherSnippetsStore`'s strategy
+ * basing on an assumption: the snippets' data is important and conflicts should
+ * be resolved instead of letting data be overwritten.
+ * */
+export function createHigherFoldersStore(
+  localStore: LocalFoldersStore,
+  remoteStore: RemoteFoldersStore,
+): LocalFoldersStore {
+  return {
+    subscribe: localStore.subscribe,
+    async upsert(folder: DisplayFolder): Promise<void> {
+      await localStore.upsert(folder)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.upsert(folder)
+      }
+    },
+    async delete(id: string): Promise<void> {
+      await localStore.delete(id)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.delete(id)
+      }
+    },
+  }
+}
+
+export function reloadRemoteFoldersStore(
+  localStore: LocalFoldersStore,
+  remoteStore: RemoteFoldersStore,
+  intervalMs: number,
+) {
+  const stateStore = createFolderStateMap(localStore, remoteStore)
+  setInterval(
+    async () => {
+      await remoteStore.refresh()
+      for (const [id, state] of Object.entries(get(stateStore))) {
+        if (state === 'remote-only') {
+          const remoteFolder = get(stateStore.remoteMap)[id]
+          await localStore.upsert(remoteFolder)
+        }
+        else if (state === 'local-only') {
+          const localFolder = get(stateStore.localMap)[id]
+          await remoteStore.upsert(localFolder)
+        }
```

Comment:
On "real-life" testing I encountered a bug with this.

Context: Suppose I open Crypta on my desktop and call it "client A". I also open Crypta on my laptop and call it "client B". Both connect to the same `ws4sqlite` server.

Reproducing steps:

- Client A create a new folder
- Data gets synchronized to the server correctly
- Client B "sees" the folder, but with "Untitled" as the name

Client A's view:

![image](https://github.com/thanhnguyen2187/crypta/assets/27758849/4ab4b2ed-d370-4d99-b56d-5e2d58587cca)

Client B's view:

![image](https://github.com/thanhnguyen2187/crypta/assets/27758849/c6e25370-55b0-45f5-9e2f-df4d48b1d609)